packages/grafana-ui/src/components/Table/TableNG/utils.test.ts

```diff
@@ -0,0 +1,1731 @@
+      const field: Field = {
+        name: 'test',
+        type: FieldType.string,
+        config: {},
+        values: [],
+      };
+
+      const options = getCellOptions(field);
+
+      // Check that default options are returned
+      expect(options).toEqual({ type: TableCellDisplayMode.Auto });
+    });
+
+    it('should extract cell options from field config', () => {
+      const field: Field = {
+        name: 'test',
+        type: FieldType.string,
+        config: {
+          custom: {
+            cellOptions: {
+              type: TableCellDisplayMode.ColorText,
+              inspectEnabled: false,
+              wrapText: true,
+            },
+          },
+        },
+        values: [],
+      };
+
+      const options = getCellOptions(field);
+
+      expect(options).toEqual({
+        type: TableCellDisplayMode.ColorText,
+        inspectEnabled: false,
+        wrapText: true,
+      });
+    });
+
+    it('should handle legacy displayMode property', () => {
+      const field: Field = {
+        name: 'test',
+        type: FieldType.string,
+        config: {
+          custom: {
+            displayMode: 'color-background',
+          },
+        },
+        values: [],
+      };
+
+      const options = getCellOptions(field);
+
+      // The legacy displayMode should be converted to the new format
+      expect(options.type).toBe(TableCellDisplayMode.ColorBackground);
+    });
+
+    it('should prioritize cellOptions over legacy displayMode', () => {
+      const field: Field = {
+        name: 'test',
+        type: FieldType.string,
+        config: {
+          custom: {
+            displayMode: 'color-background',
+            cellOptions: {
+              type: TableCellDisplayMode.ColorText,
+            },
+          },
+        },
+        values: [],
+      };
+
+      const options = getCellOptions(field);
+
+      expect(options.type).toBe(TableCellDisplayMode.ColorBackground);
+    });
+
+    it('should handle image display mode', () => {
+      const field: Field = {
+        name: 'test',
+        type: FieldType.string,
+        config: {
+          custom: {
+            cellOptions: {
+              type: TableCellDisplayMode.Image,
+              // Add image-specific options if they exist
+            },
+          },
+        },
+        values: [],
+      };
+
+      const options = getCellOptions(field);
+
+      expect(options.type).toBe(TableCellDisplayMode.Image);
+    });
+
+    it('should handle JSON display mode', () => {
+      const field: Field = {
+        name: 'test',
+        type: FieldType.string,
+        config: {
+          custom: {
+            cellOptions: {
+              type: TableCellDisplayMode.JSONView,
+            },
+          },
+        },
+        values: [],
+      };
+
+      const options = getCellOptions(field);
+
+      expect(options.type).toBe(TableCellDisplayMode.JSONView);
+    });
+  });
+
+  describe('getCellLinks', () => {
```

Comment:
@adela-almasan can you please give this another look? I'm not 100% confident on the coverage here.